addons/l10n_tw_edi_ecpay/tests/test_edi.py

```diff
@@ -0,0 +1,348 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from datetime import datetime
+from freezegun import freeze_time
+from urllib.parse import urljoin
+
+from odoo.addons.account.tests.test_account_move_send import TestAccountMoveSendCommon
+from odoo.addons.l10n_tw_edi_ecpay.controllers.main import EcpayInvoiceController
+from odoo import http
+from odoo.exceptions import UserError
+from odoo.tests import tagged
+from odoo.tests.common import HttpCase
+from unittest.mock import patch
+
+CALL_API_METHOD = 'odoo.addons.l10n_tw_edi_ecpay.utils.EcPayAPI.call_ecpay_api'
+
+
+@tagged('post_install_l10n', 'post_install', '-at_install')
+class L10nTWITestEdi(TestAccountMoveSendCommon, HttpCase):
+    def setUp(self):
+        super().setUp()
+        self.EcpayInvoiceController = EcpayInvoiceController()
```

Comment:
Do you really need to create the controller yourself? And if so, can't you do it in setupclass?